src/policy/policy.cpp

```diff
@@ -185,9 +185,16 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
     return true;
 }
 
+namespace Policy {
+    static int32_t blockSizeAcceptLimitCached = -1;
+}
+
 int32_t Policy::blockSizeAcceptLimit()
 {
-    int limit = -1;
+    if (blockSizeAcceptLimitCached != -1)
+        return blockSizeAcceptLimitCached;
+
+    int32_t &limit = blockSizeAcceptLimitCached;
```

Comment:
I didn't benchmark this, but I know that the mapArgs is most of the time mostly empty (its a map, any amount of items below 100 is "empty").

The atof may be slightly expensive, not sure if its worth it to cache this.